### PR TITLE
SC2: make apworld compatible

### PR DESCRIPTION
**Pull request by Berserker66**
Please format your title with what portion of the project this pull request is
targeting and what it's changing.

ex. "MyGame4: implement new game" or "Docs: add new guide for customizing MyGame3"

## What is this fixing or adding?
makes it so SC2 can generate as apworld, though client will currently break. As noted in the 0.3.5 release, Client support is still WIP.

## How was this tested?
running Generate after zipping the world and moving the source folder out

### CODE REVIEW

File: worlds/sc2wol/LogicMixin.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 from BaseClasses import MultiWorld
-from ..AutoWorld import LogicMixin
```
Comment (black-sliver) [inline worlds/sc2wol/LogicMixin.py:2]: Should probably update those imports in the world api.md and all worlds, so it's less work to make it apworld-compatible.

File: worlds/sc2wol/Regions.py [hunk 2]
```diff
@@ -1,8 +1,8 @@
-from typing import List, Set, Dict, Tuple, Optional, Callable, NamedTuple
+from typing import List, Set, Dict, Tuple, Optional, Callable
 from BaseClasses import MultiWorld, Region, Entrance, Location, RegionType
 from .Locations import LocationData
 from .Options import get_option_value
-from worlds.sc2wol.MissionTables import MissionInfo, vanilla_shuffle_order, vanilla_mission_req_table, \
+from .MissionTables import MissionInfo, vanilla_shuffle_order, vanilla_mission_req_table, \
```
Comment (black-sliver) [inline worlds/sc2wol/Regions.py:5]: and maybe also add a note to world api.md that you are supposed to do relative imports *within* the world

**Review by black-sliver — COMMENTED** [review]
I'll do the actual testing/approval later today. Just wanted to leave the comments below before I forget

**Review by black-sliver — CHANGES_REQUESTED** [review]
Starcraft2Client does not work with the apworld.
`from worlds.sc2wol import SC2WoLWorld` fails in line 25.

**Comment by Berserker66** [comment]
Correct. I'm not sure yet how I want to go about it. currently imports only work through the custom hook so you can't run an import from base into a world directly. I believe importlib supports adding a Loader to support this, but I'd have to dig into that for a while.

**Comment by black-sliver** [comment]
Do you want to merge that in two parts? If so, should leave a comment somewhere.

One way to do the import in sc2client, while ugly, would be
```python
import sys
from worlds import AutoWorldRegister
sc2wol = sys.modules[AutoWorldRegister.world_types["Starcraft 2 Wings of Liberty"].__module__]
SC2WOL_LOC_ID_OFFSET = sc2wol.Locations.SC2WOL_LOC_ID_OFFSET
```
instead of
```python
from worlds.sc2wol.Locations import SC2WOL_LOC_ID_OFFSET
```

**Review by black-sliver — APPROVED** [review]
From what i can tell, sc2 apworld was successfully tested in discord. Since #1069 client also works. I fixed the import order similar to SNIClient.

The one remaining problem is that we don't get sc2ap and nest_asyncio when running from apworld. I did not add apworld support to #1070 because apworlds are supposed to be shipped to non-pip-enabled installs. Adding apworld support to ModuleUpdate would potentially hide issues.

One solution would be to add sc2ap and nest_asyncio to core /requirements.txt, but that way we can't pin versions in the world.

**Pull request merged**